--- ruby-1.6.5/configure	2001-10-13 12:26:38+09	1.1
+++ ruby-1.6.5/configure	2001-10-13 12:27:07+09
@@ -4733,7 +4733,7 @@
 			LDFLAGS=""
 			DLDFLAGS="$ARCH_FLAG"
 			rb_cv_dlopen=yes ;;
-	darwin*)	LDSHARED='cc -dynamic -bundle -undefined suppress'
+	darwin*)	LDSHARED='cc -dynamic -bundle -undefined suppress -flat_namespace'
 			LDFLAGS=""
 			DLDFLAGS="$ARCH_FLAG"
 			rb_cv_dlopen=yes ;;
@@ -5266,7 +5266,7 @@
 	;;
     darwin*)
 	LIBRUBY_SO='lib$(RUBY_INSTALL_NAME).$(MAJOR).$(MINOR).$(TEENY).dylib'
-	LIBRUBY_LDSHARED='cc -dynamiclib -undefined suppress'
+	LIBRUBY_LDSHARED='cc -dynamiclib -undefined suppress -flat_namespace'
 	LIBRUBY_DLDFLAGS='-install_name lib$(RUBY_INSTALL_NAME).dylib -current_version $(MAJOR).$(MINOR).$(TEENY) -compatibility_version $(MAJOR).$(MINOR)'
 	LIBRUBY_ALIASES='lib$(RUBY_INSTALL_NAME).$(MAJOR).$(MINOR).dylib lib$(RUBY_INSTALL_NAME).dylib'
 	;;
--- ruby-1.6.5/configure.in	2001-10-13 12:23:22+09	1.1
+++ ruby-1.6.5/configure.in	2001-10-13 12:25:41+09
@@ -582,7 +582,7 @@
 			LDFLAGS=""
 			DLDFLAGS="$ARCH_FLAG"
 			rb_cv_dlopen=yes ;;
-	darwin*)	LDSHARED='cc -dynamic -bundle -undefined suppress'
+	darwin*)	LDSHARED='cc -dynamic -bundle -undefined suppress -flat_namespace'
 			LDFLAGS=""
 			DLDFLAGS="$ARCH_FLAG"
 			rb_cv_dlopen=yes ;;
@@ -864,7 +864,7 @@
 	;;
     darwin*)
 	LIBRUBY_SO='lib$(RUBY_INSTALL_NAME).$(MAJOR).$(MINOR).$(TEENY).dylib'
-	LIBRUBY_LDSHARED='cc -dynamiclib -undefined suppress'
+	LIBRUBY_LDSHARED='cc -dynamiclib -undefined suppress -flat_namespace'
 	LIBRUBY_DLDFLAGS='-install_name lib$(RUBY_INSTALL_NAME).dylib -current_version $(MAJOR).$(MINOR).$(TEENY) -compatibility_version $(MAJOR).$(MINOR)'
 	LIBRUBY_ALIASES='lib$(RUBY_INSTALL_NAME).$(MAJOR).$(MINOR).dylib lib$(RUBY_INSTALL_NAME).dylib'
 	;;
